agent/src/main/java/com/cloud/agent/Agent.java

```diff
@@ -312,7 +313,7 @@ public void start() {
     }
 
     public void stop(final String reason, final String detail) {
-        logger.info("Stopping the agent: Reason = " + reason + (detail != null ? ": Detail = " + detail : ""));
+        logger.info("Stopping the agent: Reason = {}", reason + ObjectUtils.defaultIfNull(": Detail = " + detail, ""));
```

Comment:
This defaultIfNull does not make sense, `": Detail = " + detail` will never be null. Something like this makes more sense, but maybe using string format would be better for the second parameter.
```suggestion
        logger.info("Stopping the agent: Reason = {} {}", reason, () -> ": Detail = "  + ObjectUtils.defaultIfNull(detail, ""));
```